File: src/jvm/package/src/doc-examples/java/questpdf/docexamples/DefaultTextStyleExamples.java
package questpdf.docexamples;

import com.questpdf.fluent.Document;
import com.questpdf.helpers.Colors;
import com.questpdf.helpers.PageSize;
import com.questpdf.infrastructure.ImageCompressionQuality;
import com.questpdf.infrastructure.ImageFormat;
import com.questpdf.infrastructure.ImageGenerationSettings;
import org.junit.jupiter.api.Test;

public class DefaultTextStyleExamples extends DocExample {

    @Test
    public void example() {
        var settings = new ImageGenerationSettings();
        settings.setImageFormat(ImageFormat.Webp);
        settings.setImageCompressionQuality(ImageCompressionQuality.VeryHigh);
        settings.setRasterDpi(144);

        Document
            .create(document -> {
                document.page(page -> {
                    page.minSize(new PageSize(0f, 0f));
                    page.maxSize(new PageSize(1000f, 1000f));
                    page.defaultTextStyle(style -> style.fontSize(20f));

                    page.content()
                        .width(400f)
                        .padding(25f)
                        .defaultTextStyle(style -> style.bold().underline())
                        .column(column -> {
                            column.spacing(10f);

                            column.item().text("Inherited bold and underline");
                            column.item().text("Disabled underline, inherited bold and adjusted font color").underline(false).fontColor(Colors.Green.getDarken2());

                            column.item()
                                .defaultTextStyle(style -> style.decorationWavy().fontColor(Colors.LightBlue.getDarken3()))
                                .text("Changed underline type and adjusted font color");
                        });
                });
            })
            .generateImages(index -> output("default-text-style.webp"), settings);
    }
}

File: src/jvm/package/src/doc-examples/kotlin/questpdf/docexamples/DefaultTextStyleExamples.kt
package questpdf.docexamples

import com.questpdf.fluent.Document
import com.questpdf.helpers.Colors
import com.questpdf.helpers.PageSize
import com.questpdf.infrastructure.ImageCompressionQuality
import com.questpdf.infrastructure.ImageFormat
import com.questpdf.infrastructure.ImageGenerationSettings
import org.junit.jupiter.api.Test

class DefaultTextStyleExamples : DocExample() {

    @Test
    fun example() {
        Document
            .create {
                page {
                    minSize(PageSize(0f, 0f))
                    maxSize(PageSize(1000f, 1000f))
                    defaultTextStyle { fontSize(20f) }

                    content()
                        .width(400f)
                        .padding(25f)
                        .defaultTextStyle { bold().underline() }
                        .column {
                            spacing(10f)

                            item().text("Inherited bold and underline")
                            item().text("Disabled underline, inherited bold and adjusted font color").underline(false).fontColor(Colors.Green.Darken2)

                            item()
                                .defaultTextStyle { decorationWavy().fontColor(Colors.LightBlue.Darken3) }
                                .text("Changed underline type and adjusted font color")
                        }
                }
            }
            .generateImages({ output("default-text-style.webp") }, ImageGenerationSettings().apply {
                imageFormat = ImageFormat.Webp
                imageCompressionQuality = ImageCompressionQuality.VeryHigh
                rasterDpi = 144
            })
    }
}
